src/server/game/BattlePets/BattlePetMgr.cpp

```diff
@@ -169,7 +180,7 @@ uint8 BattlePetMgr::GetDefaultPetQuality(uint32 species)
 {
     auto itr = _defaultQualityPerSpecies.find(species);
     if (itr == _defaultQualityPerSpecies.end())
-        return 0; // default poor
+        return AsUnderlyingType(BattlePetBreedQuality::Poor); // Default
```

Comment:
Why don't you make this return `BattlePetBreedQuality` instead?

Reply:
I've changed the return type to BattlePetBreedQuality. I don't know if you want me to extend it to other fields like WorldPackets::BattlePet::BattlePet.Quality